File: ruby/lib/okapi/ld_proofs.rb
# frozen_string_literal: true

module Okapi
  # LdProofs module
  module LdProofs
    def self.create(request)
      Okapi.verify_type(request, Okapi::Proofs::V1::CreateProofRequest)
      Okapi.ffi_call('ldproofs_create_proof', request, Okapi::Proofs::V1::CreateProofResponse)
    end
  end
end

File: go/okapi/ldproofs.go
package okapi

import "github.com/trinsic-id/okapi/go/okapi/proofs/v1/proofs"

// LdProofer implements Linked-Data Proofs
type LdProofer interface {
	CreateProof(request *proofs.CreateProofRequest) (*proofs.CreateProofResponse, error)
}

// LdProofs implements Linked-Data Proofs
func LdProofs() LdProofer {
	return &ldProofs{}
}

type ldProofs struct{}

func (l *ldProofs) CreateProof(request *proofs.CreateProofRequest) (*proofs.CreateProofResponse, error) {
	response := proofs.CreateProofResponse{}
	err := callOkapiNative(request, &response, "ldproofs_create_proof")
	return &response, err
}
